### PR TITLE
Testing P2 - Start of Node System - Code Only

### PR DESCRIPTION
**Pull request by okariJC**

### CODE REVIEW

File: Assets/Local/Scripts/Testing/TestNodeGroup.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        [SerializeField]
+        private TestNode[] testNodes;
+
+        public IEnumerator Execute()
+        {
+            for (var index = 0; index < testNodes.Length; index++)
+            {
+                var node = testNodes[index];
+
+                if (node is IExecute)
```
Comment (chrissJC) [inline Assets/Local/Scripts/Testing/TestNodeGroup.cs]: This is a little gnarly for a single method.  Maybe share a common interface?  `IExecutable`, extended by `IAsyncExecutable` and `ISyncExecutable` / `Immediate` / `Instant`?

`IExecutable` could then have the `Execute` method.

Comment (okariJC) [reply Assets/Local/Scripts/Testing/TestNodeGroup.cs]: I'm actually not sure what your saying.....

Comment (chrissJC) [reply Assets/Local/Scripts/Testing/TestNodeGroup.cs]: Mostly just that it looks a little cluttered for a single method.  :)  The rest was just a suggestion, though I'm not sure how difficult that would be.

Comment (okariJC) [reply Assets/Local/Scripts/Testing/TestNodeGroup.cs]: Based on our 1 on 1 Conversation. This section reads weird...but now you understand that IExecute.Execute returns void, and IAsyncExecute.Execute return an IEnumerator...

However I will method out some of the logic to make it look cleaner

**Pull request merged**